### PR TITLE
Mining Overhaul 1.0 (aka cursed lavaland)

### PR DESCRIPTION
**Pull request by SmArtKar**
## About The Pull Request

It's a giant mining overhaul with a lot of new content added. 

## Why It's Good For The Game

It allows long mining rounds and also NO INSTAKILL BY CHASMS

## Changelog
:cl:
add: New mining mob: Crazy miner! It will have a random suit and a random mining weapon!
add: A new mining Z level called "Abyss".
add: 3 new abyss monsters: Fire Wisps, Abyssal Demons and Ash Whelps.
add: A new upgraded KC(Can be dropped by crazy miners with KC)
add: A new versions of Bubblegum, Ash Drake and Colossus. (Enraged ones) Avalible at Abyss
add: A lot of new KA modkits that will drop from megafauna
tweak: Removed slaughterlings and replaced them with rending
imageadd: Resprited SEVA and EXO suits
add: A lot of new mining loot!
add: Two new bosses: Corrupt Process and The Gladiator
/:cl:

### CODE REVIEW

**Comment by SmArtKar** [comment]
WTF i didnt press create

**Comment by ghost** [comment]
have fun reading through every single code change because there's no accurate changelog and no other maintainers to merge or deny this unatomized pr @silicons 


note: Citadel-Station-13 Main is a gameplay server. Therefore, things such as combat are permitted. I know this may be hard to understand, coming from the citadel downstream that doesn't allow combat, but because of the massive difference in design policy, it has to be taken into account that combat is a possibility that can happen in your round.

File: code/game/objects/items/kitchen.dm [hunk 1]
```diff
@@ -174,6 +174,12 @@
 	throwforce = 15
 	custom_materials = null
 
+/obj/item/kitchen/knife/combat/bone/throwing
+	name = "throwing bone dagger"
+	icon = 'icons/obj/lavaland/artefacts.dmi'
+	force = 5
+	throwforce = 15
```
Comment (ghost) [inline code/game/objects/items/kitchen.dm:181]: ??

File: code/modules/mining/equipment/explorer_gear.dm [hunk 2]
```diff
@@ -185,3 +198,48 @@
...
+
+/obj/item/clothing/head/hooded/explorer/exo/standard
+
+/obj/item/clothing/suit/hooded/explorer/exo/standard/Initialize()
+	. = ..()
+	AddComponent(/datum/component/armor_plate)
+
+/obj/item/clothing/head/hooded/explorer/exo/standard/Initialize()
+	. = ..()
+	AddComponent(/datum/component/armor_plate)
```
Comment (ghost) [inline code/modules/mining/equipment/explorer_gear.dm:213]: ???

File: code/modules/mining/equipment/explorer_gear.dm [hunk 3]
```diff
@@ -158,6 +158,19 @@
...
+
+/obj/item/clothing/head/hooded/explorer/seva/standard
+
+/obj/item/clothing/suit/hooded/explorer/seva/standard/Initialize()
+	. = ..()
+	AddComponent(/datum/component/armor_plate)
+
+/obj/item/clothing/head/hooded/explorer/seva/standard/Initialize()
+	. = ..()
+	AddComponent(/datum/component/armor_plate)
```
Comment (ghost) [inline code/modules/mining/equipment/explorer_gear.dm:172]: no the point of seva is that it's for high skill players that can dodge and don't want to deal with ash storms why are you giving it armor

File: code/modules/mining/equipment/explorer_gear.dm [hunk 4]
```diff
@@ -185,3 +198,48 @@
...
+/obj/item/clothing/suit/space/heavy_miner
+	name = "H.M.S. suit"
+	desc = "Heavy Mining Space Suit is a very armored suit intended for mining and space exploration."
+	icon_state = "mining_heavy"
+	item_state = "mining_heavy"
+	max_heat_protection_temperature = FIRE_IMMUNITY_MAX_TEMP_PROTECT
+	resistance_flags = FIRE_PROOF | LAVA_PROOF | ACID_PROOF | GOLIATH_RESISTANCE
+	mutantrace_variation = STYLE_DIGITIGRADE
+	slowdown = 1.5
+	armor = list("melee" = 70, "bullet" = 40, "laser" = 30, "energy" = 30, "bomb" = 50, "bio" = 100, "rad" = 100, "fire" = 100, "acid" = 100)
```
Comment (ghost) [inline code/modules/mining/equipment/explorer_gear.dm:225]: ??????????????????????????????????????? no we already have mining hardsuits....

File: code/modules/mining/equipment/kinetic_crusher.dm [hunk 5]
```diff
@@ -27,13 +27,29 @@
...
 	desc = "An integrated version of the standard kinetic crusher with a grinded down axe head to dissuade mis-use against crewmen. Deals damage equal to the standard crusher against creatures, however."
 	force = 10 //wouldn't want to give a borg a 20 brute melee weapon unemagged now would we
 	detonation_damage = 60
 	wielded = 1
 
+/obj/item/kinetic_crusher/premium
+	icon_state = "crusher_prem"
+	item_state = "crusher_prem0"
+	name = "kinetic detonator"
+	desc = "An upgraded version of proto-kinetic crusher, fitted with better destabiliser and a more powerful flashlight. Also, looks like it can be thrown much easier."
```
Comment (ghost) [inline code/modules/mining/equipment/kinetic_crusher.dm:42]: powercrepe

File: code/modules/mining/lavaland/necropolis_chests.dm [hunk 6]
```diff
@@ -200,6 +210,81 @@
...
+	icon_state = "concussive_gauntlets"
+	toolspeed = 0.1
+	strip_delay = 40
+	equip_delay_other = 20
+	cold_protection = HANDS
+	min_cold_protection_temperature = GLOVES_MIN_TEMP_PROTECT
+	heat_protection = HANDS
+	max_heat_protection_temperature = GLOVES_MAX_TEMP_PROTECT
+	resistance_flags = LAVA_PROOF | FIRE_PROOF //they are from lavaland after all
+	armor = list("melee" = 15, "bullet" = 25, "laser" = 15, "energy" = 15, "bomb" = 100, "bio" = 0, "rad" = 0, "fire" = 100, "acid" = 30) //mostly bone bracer armor
```
Comment (ghost) [inline code/modules/mining/lavaland/necropolis_chests.dm:226]: you accidentally added on bullet armor and made it its highest normal armor stat

File: code/modules/mining/lavaland/necropolis_chests.dm [hunk 7]
```diff
@@ -749,12 +854,55 @@
...
+			new /obj/item/melee/ghost_sword(src)
+		if(2)
+			new /obj/item/lava_staff(src)
+		if(3)
+			new /obj/item/book/granter/spell/sacredflame(src)
+			new /obj/item/gun/magic/wand/fireball(src)
+		if(4)
+			new /obj/item/dragons_blood(src)
+	new /obj/item/crusher_trophy/tail_spike(src)
+
```
Comment (ghost) [inline code/modules/mining/lavaland/necropolis_chests.dm:875]: lots of copy paste here....

File: code/modules/mining/lavaland/necropolis_chests.dm [hunk 8]
```diff
@@ -988,8 +1174,17 @@
 	name = "bloody bubblegum chest"
 
 /obj/structure/closet/crate/necropolis/bubblegum/crusher/PopulateContents()
-	..()
+	new /obj/item/clothing/suit/space/hostile_environment(src)
+	new /obj/item/clothing/head/helmet/space/hostile_environment(src)
 	new /obj/item/crusher_trophy/demon_claws(src)
```
Comment (ghost) [inline code/modules/mining/lavaland/necropolis_chests.dm:1179]: why the copypaste code...

File: code/modules/mining/lavaland/necropolis_chests.dm [hunk 9]
```diff
@@ -1007,6 +1202,45 @@
...
+	name = "enraged bubblegum chest"
+
+/obj/structure/closet/crate/necropolis/bubblegum/hard/PopulateContents()
+	new /obj/item/clothing/suit/space/heavy_miner(src)
+	new /obj/item/clothing/head/helmet/space/heavy_miner(src)
+	new /obj/item/borg/upgrade/modkit/shotgun(src)
+	var/loot = rand(1,3)
+	switch(loot)
+		if(1)
+			new /obj/item/blood_contract(src)
```
Comment (ghost) [inline code/modules/mining/lavaland/necropolis_chests.dm:1215]: we removed contract for a reason

File: code/modules/mining/lavaland/necropolis_chests.dm [hunk 10]
```diff
@@ -1007,6 +1202,45 @@
...
+		if(2)
+			new /obj/item/gun/ballistic/revolver/doublebarrel/super(src)
+		if(3)
+			new /obj/item/gun/magic/staff/spellblade/melee(src)
+
+/obj/item/gun/magic/staff/spellblade/melee
+	name = "sharpened spellblade"
+	desc = "A deadly combination of laziness and boodlust, this blade allows the user to dismember their enemies without all the hard work of actually swinging the sword. It has been sharpened to be much more deadly at melee range, bit lost it's capacity."
+	force = 35
```
Comment (ghost) [inline code/modules/mining/lavaland/necropolis_chests.dm:1240]: 🥴

File: code/modules/mining/lavaland/necropolis_chests.dm [hunk 11]
```diff
@@ -1071,14 +1305,94 @@
...
 /obj/structure/closet/crate/necropolis/colossus/crusher/PopulateContents()
-	..()
+	var/list/choices = subtypesof(/obj/machinery/anomalous_crystal)
+	var/random_crystal = pick(choices)
+	new random_crystal(src)
+	new /obj/item/organ/vocal_cords/colossus(src)
+	new /obj/item/crusher_trophy/blaster_tubes(src)
+
+/obj/structure/closet/crate/necropolis/colossus/hard
+	name = "enraged colossus chest"
```
Comment (ghost) [inline code/modules/mining/lavaland/necropolis_chests.dm:1321]: copypaste...

File: code/modules/mining/lavaland/necropolis_chests.dm [hunk 12]
```diff
@@ -1071,14 +1305,94 @@
...
+/obj/item/fallen_wings
+	name = "wings of the fallen angel"
+	desc = "A pair of white, beautiful wings. They have been cut and you can't use them, but looks like they can provide decent protection."
+	force = 0
+	w_class = WEIGHT_CLASS_BULKY
+	slot_flags = ITEM_SLOT_BACK
+	throwforce = 0
+	throw_speed = 0
+
+	armor = list("melee" = 40, "bullet" = 25, "laser" = 15, "energy" = 15, "bomb" = 25, "bio" = 0, "rad" = 0)
```
Comment (ghost) [inline code/modules/mining/lavaland/necropolis_chests.dm:1351]: i have a very very bad feeling about this

File: code/modules/mining/lavaland/power_tubes.dm [hunk 13]
```diff
@@ -0,0 +1,93 @@
+//This file contains main code about ancient red liquid. The purpose of it is to charge armor, enrage bosses and maybe(?) allow you to go to another more dangerous Z.
+
+/obj/structure/lavaland/power_collector
```
Comment (ghost) [inline code/modules/mining/lavaland/power_tubes.dm:3]: don't think maintainers want argent energy cells

File: code/modules/mining/machine_vending.dm [hunk 14]
```diff
@@ -193,7 +194,7 @@
 	return ..()
 
 /obj/machinery/mineral/equipment_vendor/proc/RedeemVoucher(obj/item/mining_voucher/voucher, mob/redeemer)
-	var/items = list(	"Survival Capsule and Explorer's Webbing" = image(icon = 'icons/obj/storage.dmi', icon_state = "explorerpack"),
+	var/items = list(	"Survival Capsule and Explorer's Webbing" = image(icon = 'icons/obj/clothing/belts.dmi', icon_state = "explorer1"),
```
Comment (kappa-sama) [inline code/modules/mining/machine_vending.dm:197]: ?

File: code/modules/mining/machine_vending.dm [hunk 15]
```diff
@@ -372,19 +373,23 @@
...
 		return
 	var/drop_location = drop_location()
 	switch(selection)
 		if("Exo-suit")
-			new /obj/item/clothing/suit/hooded/explorer/exo(drop_location)
+			new /obj/item/clothing/suit/hooded/explorer/exo/standard(drop_location)
 			new /obj/item/clothing/mask/gas/exo(drop_location)
 		if("SEVA suit")
-			new /obj/item/clothing/suit/hooded/explorer/seva(drop_location)
+			new /obj/item/clothing/suit/hooded/explorer/seva/standard(drop_location)
```
Comment (ghost) [inline code/modules/mining/machine_vending.dm:388]: 1) we can see the shitcode
2) why

File: code/modules/mining/parachute_and_ropes.dm [hunk 16]
```diff
@@ -0,0 +1,50 @@
+/obj/item/parachute
```
Comment (ghost) [inline code/modules/mining/parachute_and_ropes.dm:1]: delete this. shitcode and we already have jaunters.

File: code/modules/mob/living/simple_animal/friendly/farm_animals.dm [hunk 17]
```diff
@@ -285,7 +285,7 @@
 	turns_per_move = 3
 	butcher_results = list(/obj/item/reagent_containers/food/snacks/meat/slab/chicken = 2)
 	var/egg_type = /obj/item/reagent_containers/food/snacks/egg
-	var/food_type = /obj/item/reagent_containers/food/snacks/grown/wheat
+	var/egg_food_type = /obj/item/reagent_containers/food/snacks/grown/wheat
```
Comment (ghost) [inline code/modules/mob/living/simple_animal/friendly/farm_animals.dm:288]: ?

File: code/modules/mob/living/simple_animal/hostile/megafauna/bubblegum_enraged.dm [hunk 18]
```diff
@@ -0,0 +1,141 @@
+/mob/living/simple_animal/hostile/megafauna/bubblegum/hard
+	name = "enraged bubblegum"
+	desc = "In what passes for a hierarchy among slaughter demons, this one is king. It looks VERY enraged!"
```
Comment (ghost) [inline code/modules/mob/living/simple_animal/hostile/megafauna/bubblegum_enraged.dm:3]: why

File: code/modules/mob/living/simple_animal/hostile/megafauna/colossus_enraged.dm [hunk 19]
```diff
@@ -0,0 +1,135 @@
+/mob/living/simple_animal/hostile/megafauna/colossus/hard
+	name = "enraged colossus"
```
Comment (ghost) [inline code/modules/mob/living/simple_animal/hostile/megafauna/colossus_enraged.dm:2]: why

File: code/modules/mob/living/simple_animal/hostile/megafauna/drake_enraged.dm [hunk 20]
```diff
@@ -0,0 +1,330 @@
+/mob/living/simple_animal/hostile/megafauna/dragon/hard
+	name = "enraged ash drake"
+	desc = "A very enraged guardian of the necropolis. Better run."
```
Comment (ghost) [inline code/modules/mob/living/simple_animal/hostile/megafauna/drake_enraged.dm:3]: why

File: code/modules/mining/equipment/explorer_gear.dm (hunk 4 above)
Comment (MrJWhit) [reply code/modules/mining/equipment/explorer_gear.dm:225]: The hell, 100% rad resistance? The CE's suit doesn't even have that much.

File: code/modules/mob/living/simple_animal/hostile/megafauna/gladiator.dm [hunk 21]
```diff
@@ -0,0 +1,334 @@
...
+		say("You are not welcome into the Necropolisss.")
+		introduced |= victim
+
+/mob/living/simple_animal/hostile/megafauna/gladiator/proc/chargeattack(atom/victim, var/range)
+	face_atom(target)
+	visible_message("<span class='boldwarning'>[src] lifts his shield, and prepares to charge!</span>")
+	animate(src, color = "#ff6666", 3)
+	sleep(4)
+	face_atom(target)
+	speed = -1
```
Comment (ghost) [inline code/modules/mob/living/simple_animal/hostile/megafauna/gladiator.dm:151]: is this a thing

File: code/modules/mob/living/simple_animal/hostile/megafauna/megafauna.dm [hunk 22]
```diff
@@ -48,6 +48,12 @@
 	var/list/attack_action_types = list()
 	var/small_sprite_type
 
+	//Cita changes(Abyss)
```
Comment (ghost) [inline code/modules/mob/living/simple_animal/hostile/megafauna/megafauna.dm:51]: why

File: code/modules/mob/living/simple_animal/hostile/mining_mobs/ash_whelp.dm [hunk 23]
```diff
@@ -0,0 +1,50 @@
+/mob/living/simple_animal/hostile/asteroid/ash_whelp
```
Comment (ghost) [inline code/modules/mob/living/simple_animal/hostile/mining_mobs/ash_whelp.dm:1]: make it a subtype of the ash whelps we ALREADY have

File: code/modules/mob/living/simple_animal/hostile/mining_mobs/crazy_miners.dm [hunk 24]
```diff
@@ -0,0 +1,151 @@
...
+	switch(weapon_type)
+		if("KA")
+			ranged = 1
+			projectiletype = /obj/item/projectile/kinetic
+			loot += /obj/item/gun/energy/kinetic_accelerator
+		if("KC")
+			melee_damage_lower = 20
+			melee_damage_upper = 20 //Its KC and you are the prey
+			if(prob(15))
+				loot += /obj/item/kinetic_crusher/premium
```
Comment (ghost) [inline code/modules/mob/living/simple_animal/hostile/mining_mobs/crazy_miners.dm:77]: powercrepe

File: code/modules/mob/living/simple_animal/hostile/mining_mobs/hivelord.dm [hunk 25]
```diff
@@ -309,7 +309,7 @@
 			gloves = /obj/item/clothing/gloves/color/black
 			mask = /obj/item/clothing/mask/gas/explorer
 			if(prob(20))
-				suit = pickweight(list(/obj/item/clothing/suit/hooded/explorer/standard = 6, /obj/item/clothing/suit/hooded/cloak/goliath = 2, /obj/item/clothing/suit/hooded/explorer/exo = 6, /obj/item/clothing/suit/hooded/explorer/seva = 6))
+				suit = pickweight(list(/obj/item/clothing/suit/hooded/explorer/standard = 6, /obj/item/clothing/suit/hooded/cloak/goliath = 2, /obj/item/clothing/suit/hooded/explorer/exo/standard = 6, /obj/item/clothing/suit/hooded/explorer/seva/standard = 6))
```
Comment (ghost) [inline code/modules/mob/living/simple_animal/hostile/mining_mobs/hivelord.dm:312]: why

File: code/modules/projectiles/guns/energy/kinetic_accelerator.dm [hunk 26]
```diff
@@ -33,6 +33,7 @@
 	lefthand_file = 'icons/mob/inhands/weapons/guns_lefthand.dmi'
 	righthand_file = 'icons/mob/inhands/weapons/guns_righthand.dmi'
 	ammo_type = list(/obj/item/ammo_casing/energy/kinetic/premium)
+	max_mod_capacity = 110
```
Comment (ghost) [inline code/modules/projectiles/guns/energy/kinetic_accelerator.dm:36]: why

File: code/modules/projectiles/guns/energy/kinetic_accelerator.dm [hunk 27]
```diff
@@ -635,3 +636,139 @@
...
+		var/datum/status_effect/stacking/saw_bleed/B = L.has_status_effect(STATUS_EFFECT_SAWBLEED)
+		if(!B)
+			L.apply_status_effect(STATUS_EFFECT_SAWBLEED, modifier)
+		else
+			B.add_stacks(modifier)
+
+//Bubblegum
+/obj/item/borg/upgrade/modkit/shotgun
+	name = "shotgun blast modification kit"
+	desc = "Makes you fire 3 kinetic shots instead of one."
```
Comment (ghost) [inline code/modules/projectiles/guns/energy/kinetic_accelerator.dm:662]: why

File: code/modules/projectiles/guns/energy/kinetic_accelerator.dm [hunk 28]
```diff
@@ -635,3 +636,139 @@
...
+	KA.chambered = C
+
+/obj/item/borg/upgrade/modkit/shotgun/modify_projectile(obj/item/projectile/kinetic/K)
+	K.damage /= 2
+
+//Ash Drake
+
+/obj/item/borg/upgrade/modkit/knockback
+	name = "knockback modification kit"
+	desc = "Makes your shots deal knockback."
```
Comment (ghost) [inline code/modules/projectiles/guns/energy/kinetic_accelerator.dm:690]: why

File: code/modules/projectiles/guns/energy/kinetic_accelerator.dm [hunk 29]
```diff
@@ -635,3 +636,139 @@
...
+	..()
+	var/mob/living/T = target
+	if(istype(T) && prob(15))
+		new /obj/effect/temp_visual/hierophant/chaser(loc, src, T, 5, FALSE) //Its slower
+
+//Collosus
+
+/obj/item/borg/upgrade/modkit/bolter
+	name = "death bolt modification kit"
+	desc = "Makes your kinetic accelerator have a <b>20%</b> chance to deal double damage."
```
Comment (ghost) [inline code/modules/projectiles/guns/energy/kinetic_accelerator.dm:722]: why

File: code/modules/projectiles/guns/energy/kinetic_accelerator.dm [hunk 30]
```diff
@@ -635,3 +636,139 @@
...
+	P.original = target
+	P.damage = 0 //Nope.
+	P.preparePixelProjectile(target, src)
+	P.fire()
+
+//Legion(the huge one, not the human one)
+
+/obj/item/borg/upgrade/modkit/skull
+	name = "skull launcher modification kit"
+	desc = "Makes your shots create a legion skull on impact."
```
Comment (ghost) [inline code/modules/projectiles/guns/energy/kinetic_accelerator.dm:762]: why

**Comment by ghost** [comment]
good thing this pr won't get merged anyways because skyrat nigga is probably too lazy to fix merge conflicts, ignoring the fact maintainers aren't going to like it

**Comment by MrJWhit** [comment]
I honestly can't tell who asked for this.

**Comment by ghost** [comment]
https://github.com/Citadel-Station-13/Citadel-Station-13/pull/12562

deathride58 requested changes on Jun 20
deathride58 left a comment
**Mass porting like this isn't acceptable without damn good reason. Features are not a good reason to mass port features in a single PR. Atomize your changes.**

**Comment by silicons** [comment]
can we put our comments in a single review i just had to scroll through 5 pages of comments that just say "why" c'mon now

**Comment by ghost** [comment]
> can we put our comments in a single review i just had to scroll through 5 pages of comments that just say "why" c'mon now

do you think i know how github works

i'll say it was intentional and part of the shittery now 😎

File: code/modules/projectiles/guns/energy/kinetic_accelerator.dm (hunk 27 above)
Comment (lolman360) [reply code/modules/projectiles/guns/energy/kinetic_accelerator.dm:662]: ok u gotta agree a shotgun ka would be fun

but. 120 brute at pb range.

Comment (Ludox235) [reply code/modules/projectiles/guns/energy/kinetic_accelerator.dm:662]: 120 fucking brute at pb range? Literally one hit crit. Bruh moment.

**Comment by Ludox235** [comment]
12 conflicting files!
Congratulations sailor, ye made a PR that'll get closed since you won't bother fixing them!

**Comment by jjpark-kb** [comment]
> good thing this pr won't get merged anyways because skyrat nigga is probably too lazy to fix merge conflicts, ignoring the fact maintainers aren't going to like it

I don't know why that is relevant...?

What is relevant is that I do not see any credits in the PR from where you are porting stuff. Please give credit to Bob Joga and Investigator77 for the code and gladiator sprites respectively.

image
Uhh, okay.

**Comment by ghost** [comment]
> > good thing this pr won't get merged anyways because skyrat nigga is probably too lazy to fix merge conflicts, ignoring the fact maintainers aren't going to like it
> 
> I don't know why that is relevant...?
> 
> What is relevant is that I do not see any credits in the PR from where you are porting stuff. Please give credit to Bob Joga and Investigator77 for the code and gladiator sprites respectively.
> 
> image
> Uhh, okay.

you know i have to feel bad for the skyrat maintainers sometimes

**Comment by silicons** [comment]
hello gamers I will be leaving a proper review on this PR in a day or two

Please try not to rip each other up again like we did on the dragonborg PR until then, yikes.

**Comment by Draganfrukts** [comment]
wow man didn't say anything about me making the gladiator sprites uncool :(

**Comment by SarmentiCampbell** [comment]
Did you really rip it off from Skyrat without giving peeps credits? That kinda cringe, bro

**Comment by silicons** [comment]
hello yes still no time to do a full review on this but dropping in to say to leave credit where due.

this is very obviously a skyrat port and while people can yell all they want about the license not technically requiring this i think it should be common courtesy to mention that a good amount of this is a port and where it's ported from, rather than staying silent and not leaving a single mention about that fact.

this pr will not be merged until this is done, regardless of review status.

**Review by deathride58 — CHANGES_REQUESTED** [review]
This is quite literally just a blind port of **EVERYTHING** to do with Skyrat's lavaland. With no forms of credit whatsoever. Not even a "hey, this is from Skyrat" in the PR desc

On top of that, this PR blatantly ignores the fact that we currently deal with issues regarding miners powergaming. Kappa-sama's review points out the majority of the obvious instances, but this PR shows very little sign of regard for balance.

As I've said in the past, blind mass ports like this are absolutely not acceptable in any way, shape or form. If you **really** want to port Skyrat's lavaland, then you can start by atomizing your PR, and **giving credit to the original authors**

**Comment by deathride58** [comment]
Yknow what, I'm just gonna make this easy on everyone and just say "fuck no" to this PR. If you **really** want these changes, then atomize the PR and try bringing them in one by one, but for the love of god, **give credit.**

Looking through your Discord history though, it looks like your intent here is to try passing this entire PR off as your own with no sign of credit to the original authors. If you try pulling this kind of shit again, then your ability to contribute to the codebase will be revoked. We aren't willing to play these games.

**Pull request closed without merging**